src/ui/pages/SidePage/components/WalletConnect/WalletConnectStageTwo.tsx

```diff
@@ -47,7 +51,15 @@ const WalletConnectStageTwo = ({
 
   const handleConnectWallet = async () => {
     try {
-      // TODO: implement connect wallet logic
+      if (selectedIdentifier && pendingDAppMeerkat) {
+        await PeerConnection.peerConnection.start(selectedIdentifier.id);
+        await PeerConnection.peerConnection.connectWithDApp(pendingDAppMeerkat);
+        const connection =
+          await Agent.agent.peerConnectionMetadataStorage.getPeerConnectionMetadata(
```

Comment:
I guess the database might not be updated in time with the correct name etc - so we will need an event emitted from the core for when we update so we can update redux too, right?

Maybe we can do this in another PR

Reply:
Yes, it will become the connection without the latest `name`, `selectedAID`, and `url`.

Reply:
DTIS-937 created - lets do it in the next PR